R/backup_planning.R

```diff
@@ -0,0 +1,73 @@
+# This file is supposed to be run to create timed backups of the MNM planning google sheet.
+# The table is dumped to a binary file.
+# Pure backup for now; we will take care of restoration later.
+
+
+source("R/functions.R")
+library(googlesheets4)
+
+#' read the Planning_v2 sheet in the planning googlesheet.
+#' @details see R/functions.R/get_planning_long()
+#' @param ss The id of the planning googlesheet
+get_planning_raw <- function(ss = gs_id()) {
+  # read the planning data
+  read_sheet(
+    ss,
+    sheet = "Planning_v2",
+    col_types = "ccccclllilccccccdddddddddddccdddddddddddcc",
+    .name_repair = "minimal"
+    )
+}
+
+
+#' Procedure to store a local backup of the google planning table.
+#' @details
+#' This will take a table and save it to disk,
+#' either as `rds` or as `csv`.
+#' @param ss a google sheet ID.
+#' @param filetype rds (binary) or csv (text).
+#' @param verbose whether or not output shall be printed.
+#' @examples local_backup(verbose = TRUE, filetype = 'rds')
+local_backup <- function(ss = gs_id(), verbose = TRUE, filetype = c('rds', 'csv')) {
```

Comment:
I actually [found a dictionary](https://www.merriam-webster.com/dictionary/backup#dictionary-entry-2) which describes "backup" as a verb "to make a copy of (a computer file or data) to protect against accidental loss or corruption". 
Yet I totally see your point and will adjust! I also adjusted "perform_sanity_checks()" on the other PR. Good practice!

Reply:
At some point, though, you have to tell me why you used `gs_id()` as a function. Does it have something to do with sourcing the "functions" while working on the other script?

Reply:
> you have to tell me why you used `gs_id()` as a function.

Good point :+1:. It is debatable for sure. It would work with a `gs_id` object as well.

The same holds for `max_y()` BTW. (Which I placed in the script only to make it directly editable)

It has to do more with general practice in case of an R package:

- in documentation of functions, the default argument `ss = gs_id()` value will appear in the Usage section. CRAN requires to always use exported functions/objects, so not allowing internal functions/objects in Usage.
- exported objects can be functions or other objects (like data frames). In practice, one will rarely find other things in a package documentation index than functions, except the documentation of example data sets (data sets can be exported from the code in `R/` or packaged as a data file in `data/`; see https://r-pkgs.org/data.html). That is my main reason to put everything inside functions.

Of course we are not creating a package here, but I deliberately pretended that would be the case!